docker/docker_client.go

```diff
@@ -42,8 +47,13 @@ type dockerClient struct {
 	transport       *http.Transport
 }
 
-// newDockerClient returns a new dockerClient instance for refHostname (a host a specified in the Docker image reference, not canonicalized to dockerRegistry)
-func newDockerClient(refHostname, certPath string, tlsVerify bool) (*dockerClient, error) {
+// NewClient returns a new Client instance for refHostname (a host a specified in the Docker image reference, not canonicalized to dockerRegistry)
+func NewClient(img, certPath string, tlsVerify bool) (Client, error) {
+	ref, _, err := parseDockerImageName(img)
```

Comment:
> I’m thinking the client should be hidden from users of the library

that's how it is now, but letting ppl specify a custom client if they wish is nice leveraging golang interfaces